doc/source/train/user-guides/results.rst

```diff
@@ -0,0 +1,114 @@
+Inspecting Training Results
+===========================
+
+The return value of your :meth:`Trainer.fit() <ray.train.trainer.BaseTrainer.fit>`
+call is a :class:`~ray.air.result.Result` object.
```

Comment:
Can you add a section that explains way to create a Result? The part that's missing right now is that you can also load a previously created result from `Result.from_path` (this might go hand-in-hand with the storage location section).